### PR TITLE
Branch week7 documentation

### LINKED ISSUE #23: Week 7: Update DG - Appendix: Requirements - Use cases
[Read for Details](https://nus-cs2103-ay2223s1.github.io/website/schedule/week7/project.html#5-update-the-dg-user-stories-glossary-nfrs-use-cases)

### PR DESCRIPTION
**Pull request by ChongCheeKaiClarence**
Update User Guide
Add placeholder image and skeletal Project Portfolio Page

### CODE REVIEW

File: docs/team/chongcheekaiclarence.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+title: Chong Chee Kai Clarence's Project Portfolio Page
+---
+
+### Project: SoConnect
+
+SoConnect is a desktop app for NUS SoC students to manage their contacts, optimised for use via a Command Line Interface (CLI) while still having the benefits of a Graphical User Interface (GUI). The user interacts with it using a CLI, and it has a GUI created with JavaFX. It is written in Java, and has about 10 kLoC.
+
+Given below are my contributions to the project.
+
+* **Code contributed**: [RepoSense link](https://github.com/AY2223S1-CS2103T-W15-1/tp/pulls?q=is%3Apr+author%3AChongCheeKaiClarence)
```
Comment (Eugene-Ong-W-X) [inline docs/team/chongcheekaiclarence.md]: The RepoSense link here is incorrect. Please refer back to the 2103T instructions. You can also refer to the other PPPs for what the correct link looks like.

File: docs/UserGuide.md [hunk 2]
```diff
@@ -112,6 +112,36 @@ Examples:
...
+Example:
+* `tag edit t/friend t/bestFriend` changes the friend tag to a bestFriend tag.
+
+### Add a Tag to a Contact: `tag add`
+
+Adds an existing tag to an existing contact.
+
+Format: `tag add n/NAME t/TAG`
+
+* `Name` is the name of the contact to add `TAG` to.
```
Comment (Eugene-Ong-W-X) [inline docs/UserGuide.md]: `Name` here should be `NAME` as per line 139

File: docs/UserGuide.md [hunk 3]
```diff
@@ -249,3 +279,6 @@ Action | Format, Examples
 **Search** | `search [n/NAME] [p/PHONE_NUMBER] ...`<br> e.g., `seach n/John Doe t/cs2103t`
 **List** | `list`
 **Help** | `help`
+**tag create** | `tag create t/TAG` <br> e.g., `tag create t/friend`
+**tag edit** | `tag edit t/TAG1 t/TAG2`  <br> e.g., `tag edit t/friend t/bestFriend`
+**tag add** | `tag add n/NAME t/TAG` <br> e.g., `tag add n/John Doe t/friend`
```
Comment (Eugene-Ong-W-X) [inline docs/UserGuide.md]: `tag create`, `tag edit`, `tag add` should be properly capitalised as per the other commands in the Command Summary

Suggestions: `Create Tag`, `Edit Tag`, `Add Tag` note that the first word is a verb

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User inputs word
+2.  SoConnect gives a selection of possible words.
+3.  User chooses the right word.
+4.  SoConnect changes inputted word to the chosen word. 
+
+    Use case ends.
+
+**Extension**
+*  1a. The word changes.
+    * 1a1. SoConnect updates the selection of possible words.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md:399]: After each extension, it should be stated how the use case progresses.

For example for this extension, it should be `Use case resumes from step 3.`

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-*{More to be added}*
+**Use case: Search a contact*
+
+**MSS**
+
+1.  User requests to search a specific word in the list
+2.  SoConnect shows a list of persons related to the word.
+
+**Extension**
+* 2a. The list is empty.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: This should be `1a.` instead as it is an alternative scenario to `2.`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
       Use case resumes at step 2.
 
-*{More to be added}*
+**Use case: Search a contact*
+
+**MSS**
+
+1.  User requests to search a specific word in the list
+2.  SoConnect shows a list of persons related to the word.
+
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md:319]: The MSS should end here but the end statement is missing

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+1.  User requests for list of tags.
+2.  SoConnect shows the list of tags.
+3.  User changes a tag.
+4.  SoConnect updates the tag.
+
+    Use case ends.
+
+**Extensions**
+
+* 2a. There are no tags.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: Again, this should be `1a.` as it is an alternative to `2.`

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -304,11 +304,106 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+  Use case ends.
+
+* 3a. There is no such current tag.
+    * 1a1. SoConnect shows an error message.
```
Comment (Eugene-Ong-W-X) [inline docs/DeveloperGuide.md]: Is this a typo? Shouldn't it be `3a1` as this extension is listed under `3a`?

**Review by Eugene-Ong-W-X — CHANGES_REQUESTED** [review]
Some fixes need to be made before merging.

**Comment by ChongCheeKaiClarence** [comment]
Made updates requested. Please check. Thanks!

**Comment by ChongCheeKaiClarence** [comment]
I made the changes. Please do help to check. Thanks!

**Review by Eugene-Ong-W-X — APPROVED** [review]

**Comment by Eugene-Ong-W-X** [comment]
LGTM

**Pull request merged**